src/controller.coffee

```diff
@@ -1948,6 +1968,15 @@ define ['droplet-helper',
 
     socket.notifyChange()
 
+  Editor::populateBlock = (block, string) ->
+    newBlock = @mode.parse(string, wrapAtRoot: false).start.next.container
+
+    if newBlock?.start?.next?
+      block.start.append newBlock.start.next
+      newBlock.end.prev.append block.end
+      block.notifyChange()
```

Comment:
I did that intentionally. This is because if the users removes everything from the block, the block itself should not disappear, only the contents should. In other words I can say that the buttons should have the capability to modify the contents of the block, not the block itself (or it's structure).
Here is a pictorial representation.
Original Block
![original](https://cloud.githubusercontent.com/assets/6389325/7104748/33740806-e114-11e4-871a-aeeb63c8bd76.png)
If contents are replaced
![replace_contents](https://cloud.githubusercontent.com/assets/6389325/7104749/422fe464-e114-11e4-9beb-b0c6b0e30e3c.png)
If entire block is replaced
![replace_block](https://cloud.githubusercontent.com/assets/6389325/7104750/4ba1114e-e114-11e4-9184-ccea6e0089bf.png)


On the other hand, we can have cases when the block should also be modified (eg: hide the - button if no children left in block) in which case we should replace the entire block. This will require modifying the parser such that is parses an empty string to something meaningful.